oximeter/db/src/native/block.rs

```diff
@@ -0,0 +1,699 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright 2024 Oxide Computer Company
+
+//! Types for working with actual blocks and columns of data.
+
+use super::Error;
+use chrono::{DateTime, Utc};
+use indexmap::IndexMap;
+use std::{
+    fmt,
+    net::{Ipv4Addr, Ipv6Addr},
+};
+use uuid::Uuid;
+
+/// A set of rows and columns.
+///
+/// This is the fundamental unit of data sent between ClickHouse and its
+/// clients. It can represent a set of columns selected from a table or the
+/// results of a query. It is not necessarily the entire set of data -- most
```

Comment:
Teensy nit, considered unimportant: I believe `--` for an em-dash is an ASCIIDoc-ism, in Markdown it''s spelled 
```suggestion
/// results of a query. It is not necessarily the entire set of data --- most
```
and `--` will render as two en-dashes.

Reply:
Haha, thanks! It's all text to me, so I often miss ligatures or rendering issues like this.